src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -127,6 +156,27 @@ public ActiveDirectoryUnixAuthenticationProvider(ActiveDirectorySecurityRealm re
 
         this.userCache = cache.getUserCache();
         this.groupCache = cache.getGroupCache();
+
+        String ldapConnectTimeout;
+        String ldapReadTimeout;
+        if (realm.environmentProperties != null) {
```

Comment:
if toMap was null safe then you could replace this if/else and simplify the code